### ISSUE TITLE
False positive import error warnings when `jobs != 1`

### ISSUE THREAD

**Issue by WilkenSteiner** [pylint-dev/pylint#9282]
### Bug description

```python
I want to run Pylint on the following PEP-420 namespace package:


└── src
    ├── namespace
    │   ├── package1
    │   │   ├── __init__.py
    │   │   ├── module1.py
    │   │   └── module2.py

```
I have a symbol `A` in `module2.py` and have two relative imports of `A`, `from .module2 import A` in both `package1.__init__` and `package1.module1`.

`setup.py`:
```
from setuptools import find_namespace_packages, setup

setup(
    name="package1",
    description="package 1",
    package_dir={"": "src"},
    packages=find_namespace_packages(where="src"),
    install_requires=["pylint"],
    python_requires=">=3.9",
)
```

Another package of the same namespace is installed in the environment of the `namespace.package1` project (its content doesn't matter, it only matters that both of them declare the namespace `namespace` implicitly).

If I set `jobs=1`, I receive no warning. If I set `jobs=0` (or any other value than 0), I get two import errors, E401 and E402.
```


### Configuration

```ini
[MASTER]
ignore-paths=^build/.*$
jobs=0
source-roots=src

[MESSAGES CONTROL]
# To single out the relevant warnings
disable=
    missing-docstring, 
    unused-import,
    too-few-public-methods,
```

### Command used

```shell
pylint --rcfile ./pylintrc  **/*.py
```

### Pylint output

```python
************* Module __init__
src/namespace/package1/__init__.py:1:0: E0401: Unable to import '__init__.module1' (import-error)
************* Module module2
src/namespace/package1/module2.py:1:0: E0402: Attempted relative import beyond top-level package (relative-beyond-top-level)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.1 (main, Jan 18 2023, 15:03:55) [GCC 12.2.0]
```
